### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by kshithesh**
added Exceptions

### CODE REVIEW

File: EventView-1/pom.xml [hunk 1]
```diff
@@ -14,28 +14,27 @@
...
-
-        <!-- https://mvnrepository.com/artifact/org.apache.tomcat/tomcat-jasper -->
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-tomcat</artifactId>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
```
Comment (ram070278) [inline EventView-1/pom.xml:42]: JPA should be removed from project

File: EventView-1/src/main/java/com/eventview/exceptions/EventTypeExceptionController.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+import java.sql.Timestamp;
+
+@ControllerAdvice
+public class EventTypeExceptionController {
+    @ExceptionHandler(value = EventTypeNotFoundException.class)
```
Comment (ram070278) [inline EventView-1/src/main/java/com/eventview/exceptions/EventTypeExceptionController.java]: what is diff than EventExceptionController , why duplicate controllerAdvice for multiple exceptions

File: EventView-1/src/main/java/com/eventview/exceptions/UserExceptionController.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+import java.sql.Timestamp;
+
+@ControllerAdvice
+public class UserExceptionController {
+    @ExceptionHandler(value = UserNotFoundException.class)
+    public ResponseEntity<ErrorResponse> userNotFoundException(UserNotFoundException exception) {
```
Comment (ram070278) [inline EventView-1/src/main/java/com/eventview/exceptions/UserExceptionController.java]: same as above what is diff this and EventExceptionController , u shd combine all in same without duplicates

File: EventView-1/src/main/java/com/eventview/repo/EventRepoImpl.java [hunk 4]
```diff
@@ -23,32 +24,47 @@
 
     @Override
     public List<EventsPayload> getAllEvens() {
-        String esql = "select e.event_id,concat(u.first_name,' ',u.last_name) as full_name, et.event_type,e.event_date from events e left join users u on e.user_id = u.user_id left join eventtypes et on et.event_type_id=e.event_type_id";
-        return jdbcTemplate.query(esql, new EventsPayloadRowMapper());
+        String sql1 = "select e.event_id,concat(u.first_name,' ',u.last_name) as full_name, et.event_type,e.event_date from events e left join users u on e.user_id = u.user_id left join eventtypes et on et.event_type_id=e.event_type_id";
```
Comment (ram070278) [inline EventView-1/src/main/java/com/eventview/repo/EventRepoImpl.java]: told multiple times avoid concat here, also when are u changing Evens to Events here and many places, name sql1 to valid format like SELECT_EVENTS

File: EventView-1/src/main/java/com/eventview/repo/EventTypeRepoImpl.java [hunk 5]
```diff
@@ -1,46 +1,63 @@
...
+			return evenTypes;
+        } catch (Exception e) {
+            throw new EventTypeNotFoundException("EventType not found with id: "+ eventTypeId);
+        }
+    }
+
+    @Override
+    public void createEventType(EvenTypes evenTypes) {
+        String sql = "INSERT INTO eventtypes (event_type_id, event_id, event_type) VALUES (?,?,?)";
+        jdbcTemplate.update(sql, evenTypes.getEventTypeId(), evenTypes.getEventid(), evenTypes.getEventType());
```
Comment (ram070278) [inline EventView-1/src/main/java/com/eventview/repo/EventTypeRepoImpl.java]: what if insert fails

File: EventView-1/src/main/java/com/eventview/repo/EventTypeRepoImpl.java [hunk 6]
```diff
@@ -1,46 +1,63 @@
...
+    }
+
+    @Override
+    public void updateEventType(EvenTypes evenTypes) {
+        String sql = "update eventtypes set event_id=?, event_type=? where event_type_id=?";
+        jdbcTemplate.update(sql, evenTypes.getEventid(), evenTypes.getEventType(), evenTypes.getEventTypeId());
```
Comment (ram070278) [inline EventView-1/src/main/java/com/eventview/repo/EventTypeRepoImpl.java]: what if update fails

File: EventView-1/src/main/java/com/eventview/controller/EventTypesRestController.java [hunk 7]
```diff
@@ -22,78 +25,62 @@
 
     @GetMapping(path = "/types")
```
Comment (ram070278) [inline EventView-1/src/main/java/com/eventview/controller/EventTypesRestController.java]: why 'types' here n 'type' in other place

File: EventView-1/src/main/java/com/eventview/repo/EventTypeRepoImpl.java (hunk 5 above)
Comment (kshithesh) [reply EventView-1/src/main/java/com/eventview/repo/EventTypeRepoImpl.java]: exception with message created

File: EventView-1/src/main/java/com/eventview/repo/EventTypeRepoImpl.java (hunk 6 above)
Comment (kshithesh) [reply EventView-1/src/main/java/com/eventview/repo/EventTypeRepoImpl.java]: exception with message created

File: EventView-1/pom.xml (hunk 1 above)
Comment (kshithesh) [reply EventView-1/pom.xml:42]: > its works but this is not ideal way to use ControllerAdvice, send 404 where it generates not at system level. Read and understand ControllerAdvice.

fixed

File: EventView-1/src/main/java/com/eventview/repo/EventRepoImpl.java (hunk 4 above)
Comment (kshithesh) [reply EventView-1/src/main/java/com/eventview/repo/EventRepoImpl.java]: fixed except concat
will find a way and fix it

File: EventView-1/src/main/java/com/eventview/controller/EventTypesRestController.java (hunk 7 above)
Comment (kshithesh) [reply EventView-1/src/main/java/com/eventview/controller/EventTypesRestController.java]: fixed

**Pull request closed without merging**